In [1]:
import numpy as np 
import pandas as pd 

In [2]:
import json 
from sklearn import preprocessing

In [3]:
import matplotlib.pyplot as plt

In [4]:
import math as m 

In [5]:
%load_ext ipython_memory_usage

Enabling IPython Memory Usage, use %imu_start to begin, %imu_stop to end


In [6]:
%imu_start 

'IPython Memory Usage started'

In [6] used 0.4 MiB RAM in 0.12s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 144.6 MiB


### EDA de la dataFrame 'Athlete_events'

In [7]:
df_athlete=pd.read_csv('data/athlete_events.csv',sep=',')

df_athlete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282985              ZOU Jingyuan   M  23.0    1.58     NaN   
282986          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282987            ZUEV Alexander   M  25.0    1.93     NaN   
282988          ZVEREV Alexander   M  24.0    1.98     NaN   
282989              ZYKOVA Yulia   F  26.0     NaN     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282985  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282986                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282987                         ROC  ROC   Tokyo 2020  2020  Summer      Tokyo   
282988                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   
282989                         ROC  ROC   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event   Medal  
0                Basketball       Basketball Men's Basketball     NaN  
1                      Judo      Judo Men's Extra-Lightweight     NaN  
2                  Football           Football Men's Football     NaN  
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War    Gold  
4             Speed Skating  Speed Skating Women's 500 metres     NaN  
...                     ...                               ...     ...  
282985  Artistic Gymnastics               Men's Parallel Bars    Gold  
282986             Football                               Men  Silver  
282987       3x3 Basketball                               Men  Silver  
282988               Tennis                     Men's Singles    Gold  
282989             Shooting       50m Rifle 3 Positions Women  Silver  

[282990 rows x 14 columns]

In [7] used 154.7 MiB RAM in 5.31s (system mean cpu 42%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 299.3 MiB


### DataFrame dictionnary

In [8]:
df_dictionnary=pd.read_csv('data/dictionary.csv')

df_dictionnary=df_dictionnary.rename(columns={'GDP per Capita':'PIB/habitant','Code':'NOC'})

df_dictionnary.isna().sum()

Country          0
NOC              0
Population       5
PIB/habitant    25
dtype: int64

In [8] used 0.2 MiB RAM in 1.01s (system mean cpu 41%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 299.5 MiB


### On cherche les athlètes qui n'ont pas de team et ceux des îles TUVALU

In [9]:
df_athlete[df_athlete['NOC']=='UNK']


Name Sex  Age  Height  Weight     Team  NOC        Games  \
61080   Fritz Eccard   M  NaN     NaN     NaN  Unknown  UNK  1912 Summer   
130721     A. Laffen   M  NaN     NaN     NaN  Unknown  UNK  1912 Summer   

        Year  Season       City             Sport  \
61080   1912  Summer  Stockholm  Art Competitions   
130721  1912  Summer  Stockholm  Art Competitions   

                                      Event Medal  
61080   Art Competitions Mixed Architecture   NaN  
130721  Art Competitions Mixed Architecture   NaN

In [9] used 2.4 MiB RAM in 0.17s (system mean cpu 45%, single max cpu 83%), peaked 0.0 MiB above final usage, current RAM usage now 301.9 MiB


Fritez Eccard a participé dans l'équipe Suédoise et A.Laffen .... aucunes traces je sais pas qui est ce mec 

In [10]:
df_athlete[df_athlete['NOC']=='ROT'].nunique()


Name      10
Sex        2
Age        7
Height     7
Weight     9
Team       1
NOC        1
Games      1
Year       1
Season     1
City       1
Sport      3
Event     12
Medal      0
dtype: int64

In [10] used -2.0 MiB RAM in 0.15s (system mean cpu 38%, single max cpu 100%), peaked 2.1 MiB above final usage, current RAM usage now 299.9 MiB


12 participations parmi 10 athlètes qui ont participé aux JO ces dernières années sous la team des athlètes réfugiés. 

In [11]:
df_athlete=df_athlete.query(" NOC !='ROC' and NOC != 'UNK' ")

In [11] used -6.2 MiB RAM in 0.19s (system mean cpu 38%, single max cpu 100%), peaked 11.2 MiB above final usage, current RAM usage now 293.7 MiB


## Base données région 

In [12]:
df_regions=pd.read_csv('data/noc_regions.csv')

df_regions=df_regions.rename(columns={'region':'Country'})
df_regions

NOC      Country                 notes
0    AFG  Afghanistan                   NaN
1    AHO      Curacao  Netherlands Antilles
2    ALB      Albania                   NaN
3    ALG      Algeria                   NaN
4    AND      Andorra                   NaN
..   ...          ...                   ...
225  YEM        Yemen                   NaN
226  YMD        Yemen           South Yemen
227  YUG       Serbia            Yugoslavia
228  ZAM       Zambia                   NaN
229  ZIM     Zimbabwe                   NaN

[230 rows x 3 columns]

In [12] used 0.1 MiB RAM in 0.90s (system mean cpu 42%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 293.8 MiB


In [13]:
df_regions.nunique()

NOC        230
Country    206
notes       21
dtype: int64

In [13] used 0.0 MiB RAM in 0.11s (system mean cpu 47%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 293.8 MiB


Il y a des régions qui ont le même NOC 

In [14]:
duplicates=[df_regions.duplicated(subset='Country')]

In [14] used -6.1 MiB RAM in 0.11s (system mean cpu 44%, single max cpu 100%), peaked 6.1 MiB above final usage, current RAM usage now 287.8 MiB


In [15]:
df_regions[df_regions.duplicated(subset='Country')]

NOC         Country                   notes
12   AUS       Australia                     NaN
54   CZE  Czech Republic                     NaN
76   GDR         Germany                     NaN
79   GER         Germany                     NaN
81   GRE          Greece                     NaN
88   HKG           China               Hong Kong
126  MAS        Malaysia                     NaN
143  NBO        Malaysia            North Borneo
147  NFL          Canada            Newfoundland
171  RUS          Russia                     NaN
173  SAA         Germany                     NaN
185  SRB          Serbia                     NaN
197  TCH  Czech Republic                     NaN
208  TUV             NaN                  Tuvalu
210  UAR           Syria    United Arab Republic
213  UNK             NaN                 Unknown
214  URS          Russia                     NaN
222  VNM         Vietnam                     NaN
223  WIF        Trinidad  West Indies Federation
225  YEM           Yemen                     NaN
226  YMD           Yemen             South Yemen
227  YUG          Serbia              Yugoslavia
229  ZIM        Zimbabwe                     NaN

In [15] used 0.1 MiB RAM in 0.12s (system mean cpu 41%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 287.8 MiB


In [16]:
df_regions.isnull().sum()

NOC          0
Country      3
notes      209
dtype: int64

In [16] used -24.0 MiB RAM in 0.11s (system mean cpu 43%, single max cpu 100%), peaked 24.0 MiB above final usage, current RAM usage now 263.8 MiB


In [17]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [17] used -4.0 MiB RAM in 0.11s (system mean cpu 41%, single max cpu 100%), peaked 4.0 MiB above final usage, current RAM usage now 259.9 MiB


Il y a  3 des regions qui ne sont pas attribuées, ROT ( refugee olympic team)  , TUVALU ( 9 archipels qui constituent un État ) et UNK qui correspond à Unknown 

### On ajoute les iles TUVALU 

In [18]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [18] used -2.3 MiB RAM in 0.11s (system mean cpu 55%, single max cpu 100%), peaked 2.3 MiB above final usage, current RAM usage now 257.5 MiB


In [19]:
df_regions=df_regions.query(" NOC != 'ROT' and NOC != 'UNK' ")

df_regions.isnull().sum()

NOC          0
Country      1
notes      209
dtype: int64

In [19] used -3.4 MiB RAM in 0.12s (system mean cpu 49%, single max cpu 100%), peaked 3.4 MiB above final usage, current RAM usage now 254.1 MiB


In [20]:
df_regions['Country']=df_regions['Country'].fillna('Tuvalu')

In [20] used -1.1 MiB RAM in 0.11s (system mean cpu 40%, single max cpu 100%), peaked 1.1 MiB above final usage, current RAM usage now 253.0 MiB


<ipython-input-20-0f7f308fe64b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_regions['Country']=df_regions['Country'].fillna('Tuvalu')


In [21]:
df_regions[df_regions['Country']=='Tuvalu']

NOC Country   notes
208  TUV  Tuvalu  Tuvalu

In [21] used -2.1 MiB RAM in 0.12s (system mean cpu 37%, single max cpu 100%), peaked 2.1 MiB above final usage, current RAM usage now 251.0 MiB


In [22]:
df_regions.isnull().sum()

NOC          0
Country      0
notes      209
dtype: int64

In [22] used 0.0 MiB RAM in 0.11s (system mean cpu 33%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 251.0 MiB


# On va merge les dataframes 

## Merge en left 

In [23]:
df_complete=pd.merge(pd.merge(df_athlete,df_regions,on='NOC',how='left'),df_dictionnary,on='NOC',how='left')
#df_complete=df_complete.drop(labels='',axis=1)

df_complete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer      Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event   Medal  \
0                Basketball       Basketball Men's Basketball     NaN   
1                      Judo      Judo Men's Extra-Lightweight     NaN   
2                  Football           Football Men's Football     NaN   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War    Gold   
4             Speed Skating  Speed Skating Women's 500 metres     NaN   
...                     ...                               ...     ...   
282622            Taekwondo                       Women -57kg    Gold   
282623  Artistic Gymnastics                        Men's Team  Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars    Gold   
282625             Football                               Men  Silver   
282626               Tennis                     Men's Singles    Gold   

          Country_x notes      Country_y    Population  PIB/habitant  
0             China   NaN          China  1.371220e+09   8027.683810  
1             China   NaN          China  1.371220e+09   8027.683810  
2           Denmark   NaN        Denmark  5.676002e+06  51989.293471  
3           Denmark   NaN        Denmark  5.676002e+06  51989.293471  
4       Netherlands   NaN    Netherlands  1.693652e+07  44299.768085  
...             ...   ...            ...           ...           ...  
282622          USA   NaN  United States  3.214188e+08  56115.718426  
282623        China   NaN          China  1.371220e+09   8027.683810  
282624        China   NaN          China  1.371220e+09   8027.683810  
282625        Spain   NaN          Spain  4.641827e+07  25831.582305  
282626      Germany   NaN        Germany  8.141314e+07  41313.313995  

[282627 rows x 19 columns]

In [23] used 72.6 MiB RAM in 0.26s (system mean cpu 29%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 323.6 MiB


In [24]:
df_complete=df_complete.query(" NOC !='ROC' ")

In [24] used -1.6 MiB RAM in 0.15s (system mean cpu 26%, single max cpu 100%), peaked 1.6 MiB above final usage, current RAM usage now 322.1 MiB


In [25]:
#df_complete.info()

In [25] used -0.0 MiB RAM in 0.11s (system mean cpu 30%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 322.0 MiB


282 627 lignes , c'est normal on a enlevé les  réfugiés

In [26]:
df_complete.loc[(df_complete['Medal'].isna(),'Medal')]='Pas_medaille'



In [26] used -11.7 MiB RAM in 0.13s (system mean cpu 38%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 310.4 MiB


## Encoding test 

In [27]:
dictio_score_medaille={'Pas_medaille':0,
                      'Bronze':1,
                      'Silver':2,
                      'Gold':3}

df_complete['Score_medaille']=df_complete['Medal'].map(dictio_score_medaille)

df_complete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer      Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event         Medal  \
0                Basketball       Basketball Men's Basketball  Pas_medaille   
1                      Judo      Judo Men's Extra-Lightweight  Pas_medaille   
2                  Football           Football Men's Football  Pas_medaille   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War          Gold   
4             Speed Skating  Speed Skating Women's 500 metres  Pas_medaille   
...                     ...                               ...           ...   
282622            Taekwondo                       Women -57kg          Gold   
282623  Artistic Gymnastics                        Men's Team        Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars          Gold   
282625             Football                               Men        Silver   
282626               Tennis                     Men's Singles          Gold   

          Country_x notes      Country_y    Population  PIB/habitant  \
0             China   NaN          China  1.371220e+09   8027.683810   
1             China   NaN          China  1.371220e+09   8027.683810   
2           Denmark   NaN        Denmark  5.676002e+06  51989.293471   
3           Denmark   NaN        Denmark  5.676002e+06  51989.293471   
4       Netherlands   NaN    Netherlands  1.693652e+07  44299.768085   
...             ...   ...            ...           ...           ...   
282622          USA   NaN  United States  3.214188e+08  56115.718426   
282623        China   NaN          China  1.371220e+09   8027.683810   
282624        China   NaN          China  1.371220e+09   8027.683810   
282625        Spain   NaN          Spain  4.641827e+07  25831.582305   
282626      Germany   NaN        Germany  8.141314e+07  41313.313995   

        Score_medaille  
0                    0  
1                    0  
2                    0  
3                    3  
4                    0  
...                ...  
282622               3  
282623               1  
282624               3  
282625               2  
282626               3  

[282627 rows x 20 columns]

In [27] used -1.2 MiB RAM in 0.14s (system mean cpu 35%, single max cpu 100%), peaked 1.2 MiB above final usage, current RAM usage now 309.2 MiB


In [28]:
#Marche aussi 
''' 
medaille_categories=['Pas_medaille','Bronze','Silver','Gold']
encode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])

df_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])

df_complete
''' 

" \nmedaille_categories=['Pas_medaille','Bronze','Silver','Gold']\nencode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])\n\ndf_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])\n\ndf_complete\n"

In [28] used 0.0 MiB RAM in 0.11s (system mean cpu 34%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 309.2 MiB


## Analyse valeurs nulles 

In [29]:
df_complete.isna().sum()

Name                   0
Sex                  156
Age                 9628
Height             67142
Weight             74386
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Event               9278
Medal                  0
Country_x            419
notes             277510
Country_y          26221
Population         27593
PIB/habitant       35919
Score_medaille         0
dtype: int64

In [29] used 2.5 MiB RAM in 0.25s (system mean cpu 18%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 311.7 MiB


156 sports où on sait pas ce que c'est .... et des country (26 221) qui ne sont pas attribués non plus 

Pour les Country , on va merge 

In [30]:
#df_complete_test=df_complete.copy()

df_complete=pd.merge(df_complete,df_regions,on='NOC',how='left')

df_complete['Country']=df_complete['Country_y'].fillna(df_complete['Country_x'])

In [30] used 47.4 MiB RAM in 0.19s (system mean cpu 31%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 359.1 MiB


In [31]:
df_complete= df_complete.drop(['Country_x', 'Country_y'], axis=1)

In [31] used 38.2 MiB RAM in 0.16s (system mean cpu 28%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 397.2 MiB


In [32]:
liste_pays=df_regions['Country'].tolist()
liste_NOC=df_regions['NOC'].tolist()

dictio_NOC_pays=dict(zip(liste_NOC,liste_pays))

dictio_NOC_pays['SGP']='Singapour'
dictio_NOC_pays['LBN']='Liban'


df_complete['Country']=df_complete['Country'].fillna(df_complete['NOC'].map(dictio_NOC_pays))
#liste_region

#df_complete_test.isna().sum()

In [32] used -1.7 MiB RAM in 0.14s (system mean cpu 35%, single max cpu 100%), peaked 0.3 MiB above final usage, current RAM usage now 395.5 MiB


In [33]:
#test=df_complete_test[df_complete_test['Country'].isna()]

#test['Country']=test['Country'].fillna(test['NOC'].map(dictio_NOC_pays))

#test.groupby('NOC')['Name'].count().reset_index()

In [33] used 0.0 MiB RAM in 0.11s (system mean cpu 41%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 395.6 MiB


Il y a quatre NOC qui ne correspondent à aucun pays dans notre EOR : réfugiés , ROT réfugiés aussi , LBN c'est le Liban (Lebanon en anglais )

In [34]:
df_complete=df_complete.query(" NOC !='ROT' and NOC !='EOR' ")



In [34] used -27.9 MiB RAM in 0.20s (system mean cpu 37%, single max cpu 100%), peaked 32.2 MiB above final usage, current RAM usage now 367.7 MiB


In [35]:
df_complete=df_complete.drop(['notes_x','notes_y'],axis=1)

In [35] used 13.0 MiB RAM in 0.15s (system mean cpu 34%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 380.7 MiB


## Opérations sur les sports 

In [36]:
liste_sports=df_complete['Sport'].unique().tolist()

print(liste_sports)

['Basketball', 'Judo', 'Football', 'Tug-Of-War', 'Speed Skating', 'Cross Country Skiing', 'Athletics', 'Ice Hockey', 'Swimming', 'Badminton', 'Sailing', 'Biathlon', 'Gymnastics', 'Art Competitions', 'Alpine Skiing', 'Handball', 'Weightlifting', 'Wrestling', 'Luge', 'Water Polo', 'Hockey', 'Rowing', 'Bobsleigh', 'Fencing', 'Equestrianism', 'Shooting', 'Boxing', 'Taekwondo', 'Cycling', 'Diving', 'Canoeing', 'Tennis', 'Modern Pentathlon', 'Figure Skating', 'Golf', 'Softball', 'Archery', 'Volleyball', 'Synchronized Swimming', 'Table Tennis', 'Nordic Combined', 'Baseball', 'Rhythmic Gymnastics', 'Freestyle Skiing', 'Rugby Sevens', 'Trampolining', 'Beach Volleyball', 'Triathlon', 'Ski Jumping', 'Curling', 'Snowboarding', 'Rugby', 'Short Track Speed Skating', 'Skeleton', 'Lacrosse', 'Polo', 'Cricket', 'Racquets', 'Motorboating', 'Military Ski Patrol', 'Croquet', 'Jeu De Paume', 'Roque', 'Alpinism', 'Basque Pelota', 'Aeronautics', 'Cycling Road', 'Artistic Gymnastics', 'Karate', 'Trampoline Gy

In [37]:
table_sport=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sport

Sport  Name
0   3x3 Basketball    56
1      Aeronautics     1
2    Alpine Skiing  8829
3         Alpinism    25
4          Archery  2461
..             ...   ...
79      Tug-Of-War   170
80      Volleyball  3666
81      Water Polo  4119
82   Weightlifting  4128
83       Wrestling  7425

[84 rows x 2 columns]

In [37] used -29.1 MiB RAM in 0.16s (system mean cpu 69%, single max cpu 100%), peaked 29.2 MiB above final usage, current RAM usage now 312.2 MiB


In [38]:
df_complete=df_complete[df_complete['Year']>=2000]

In [38] used -12.5 MiB RAM in 0.13s (system mean cpu 57%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 299.7 MiB


In [39]:
df_complete

Name Sex   Age  Height  Weight  \
1                       A Lamusi   M  23.0  170.00    60.0   
28              Antti Sami Aalto   M  26.0  186.00    96.0   
31             Jyri Tapani Aalto   M  31.0  172.00    70.0   
33            Minna Maarit Aalto   F  34.0  159.00    55.5   
40      Juhamatti Tapio Aaltonen   M  28.0  184.00    85.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season  \
1                            China  CHN  2012 Summer  2012  Summer   
28                         Finland  FIN  2002 Winter  2002  Winter   
31                         Finland  FIN  2000 Summer  2000  Summer   
33                         Finland  FIN  2000 Summer  2000  Summer   
40                         Finland  FIN  2014 Winter  2014  Winter   
...                            ...  ...          ...   ...     ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer   
282625                       Spain  ESP   Tokyo 2020  2020  Summer   
282626                     Germany  GER   Tokyo 2020  2020  Summer   

                  City                Sport                         Event  \
1               London                 Judo  Judo Men's Extra-Lightweight   
28      Salt Lake City           Ice Hockey   Ice Hockey Men's Ice Hockey   
31              Sydney            Badminton       Badminton Men's Singles   
33              Sydney              Sailing    Sailing Women's Windsurfer   
40               Sochi           Ice Hockey   Ice Hockey Men's Ice Hockey   
...                ...                  ...                           ...   
282622           Tokyo            Taekwondo                   Women -57kg   
282623           Tokyo  Artistic Gymnastics                    Men's Team   
282624           Tokyo  Artistic Gymnastics           Men's Parallel Bars   
282625           Tokyo             Football                           Men   
282626           Tokyo               Tennis                 Men's Singles   

               Medal    Population  PIB/habitant  Score_medaille  \
1       Pas_medaille  1.371220e+09   8027.683810               0   
28      Pas_medaille  5.482013e+06  42311.036231               0   
31      Pas_medaille  5.482013e+06  42311.036231               0   
33      Pas_medaille  5.482013e+06  42311.036231               0   
40            Bronze  5.482013e+06  42311.036231               1   
...              ...           ...           ...             ...   
282622          Gold  3.214188e+08  56115.718426               3   
282623        Bronze  1.371220e+09   8027.683810               1   
282624          Gold  1.371220e+09   8027.683810               3   
282625        Silver  4.641827e+07  25831.582305               2   
282626          Gold  8.141314e+07  41313.313995               3   

              Country  
1               China  
28            Finland  
31            Finland  
33            Finland  
40            Finland  
...               ...  
282622  United States  
282623          China  
282624          China  
282625          Spain  
282626        Germany  

[96730 rows x 18 columns]

In [39] used -2.5 MiB RAM in 0.13s (system mean cpu 56%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 297.3 MiB


## Opérations sur les Ages

In [40]:
table_sport_age=df_complete.groupby('Sport')['Age'].mean().apply(m.ceil).reset_index()

table_sport_age

Sport  Age
0        3x3 Basketball   30
1         Alpine Skiing   25
2               Archery   27
3   Artistic Gymnastics   25
4     Artistic Swimming   24
..                  ...  ...
64            Triathlon   29
65           Volleyball   27
66           Water Polo   27
67        Weightlifting   25
68            Wrestling   27

[69 rows x 2 columns]

In [40] used -4.7 MiB RAM in 0.13s (system mean cpu 45%, single max cpu 100%), peaked 0.1 MiB above final usage, current RAM usage now 292.5 MiB


In [41]:
liste_sport=table_sport_age['Sport'].tolist()
liste_age=table_sport_age['Age'].tolist()

dictio_age_sport=dict(zip(liste_sport,liste_age))

In [41] used -1.2 MiB RAM in 0.11s (system mean cpu 54%, single max cpu 100%), peaked 1.2 MiB above final usage, current RAM usage now 291.4 MiB


In [42]:
df_complete['Age']=df_complete['Age'].fillna(df_complete['Sport'].map(dictio_age_sport))

df_complete.isna().sum()

Name                  0
Sex                 156
Age                 156
Height             7658
Weight            12694
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport               156
Event              9249
Medal                 0
Population         2855
PIB/habitant       5741
Score_medaille        0
Country               0
dtype: int64

In [42] used 0.8 MiB RAM in 0.16s (system mean cpu 45%, single max cpu 100%), peaked 0.1 MiB above final usage, current RAM usage now 292.2 MiB


In [43]:
df_complete[df_complete['Age'].isna()].groupby('Medal')['Name'].count().reset_index()

Medal  Name
0  Pas_medaille   156

In [43] used -2.7 MiB RAM in 0.11s (system mean cpu 75%, single max cpu 100%), peaked 2.7 MiB above final usage, current RAM usage now 289.5 MiB


In [44]:
#df_tokyo=pd.read_csv('data/athlete_events_tokyo_cleaned.csv')

#df_tokyo.isna().sum()

In [44] used -3.2 MiB RAM in 0.10s (system mean cpu 48%, single max cpu 100%), peaked 3.2 MiB above final usage, current RAM usage now 286.2 MiB


Enfaite il y a 156 personnes qui manquent prcq c'est les qu'on a pas les sport 

In [45]:
df_complete=df_complete.dropna(subset='Age')

In [45] used 11.0 MiB RAM in 0.13s (system mean cpu 61%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 297.3 MiB


In [46]:
df_complete.isna().sum()

Name                  0
Sex                   0
Age                   0
Height             7502
Weight            12538
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Event              9093
Medal                 0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [46] used -1.2 MiB RAM in 0.15s (system mean cpu 64%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 296.1 MiB


## Opérations sur les poids et les tailles 

In [47]:
df_complete.isna().sum()

Name                  0
Sex                   0
Age                   0
Height             7502
Weight            12538
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Event              9093
Medal                 0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [47] used -5.3 MiB RAM in 0.16s (system mean cpu 43%, single max cpu 100%), peaked 5.3 MiB above final usage, current RAM usage now 290.8 MiB


In [48]:
table_taille=df_complete[df_complete['Height'].isna()==False].groupby(['Sport'])['Height'].mean().apply(m.ceil).reset_index()

liste_sport=table_taille['Sport'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_taille=table_taille['Height'].tolist()
print(table_taille)
liste_sport_sex=[]
for i in range(len(liste_sport)):
    liste_sport_sex.append((liste_sport[i],liste_taille[i]))
#print(liste_sport_sex)

dictio_sport_sex=dict(zip(liste_sport,liste_taille))



                  Sport  Height
0        3x3 Basketball       2
1         Alpine Skiing     176
2               Archery     174
3   Artistic Gymnastics       2
4     Artistic Swimming       2
..                  ...     ...
62            Triathlon     169
63           Volleyball     161
64           Water Polo     169
65        Weightlifting     159
66            Wrestling     171

[67 rows x 2 columns]
In [48] used -30.7 MiB RAM in 0.14s (system mean cpu 34%, single max cpu 100%), peaked 30.7 MiB above final usage, current RAM usage now 260.1 MiB


In [49]:
df_complete['Height'] = df_complete['Height'].fillna(df_complete['Sport'].map(dictio_sport_sex))


df_complete.isna().sum()

<ipython-input-49-30982cdfd39c>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_complete['Height'] = df_complete['Height'].fillna(df_complete['Sport'].map(dictio_sport_sex))


Name                  0
Sex                   0
Age                   0
Height              159
Weight            12538
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Event              9093
Medal                 0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [49] used -24.5 MiB RAM in 0.16s (system mean cpu 50%, single max cpu 100%), peaked 24.5 MiB above final usage, current RAM usage now 235.6 MiB


In [50]:
#df_complete[df_complete['Height'].isna()]

In [50] used 0.0 MiB RAM in 0.10s (system mean cpu 42%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 235.7 MiB


In [51]:
table_poids=df_complete[df_complete['Weight'].isna()==False].groupby(['Sport'])['Weight'].mean().apply(m.ceil).reset_index()

liste_sport_2=table_poids['Sport'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids=table_poids['Weight'].tolist()



dictio_sport_poids=dict(zip(liste_sport_2,liste_poids))

In [51] used -3.6 MiB RAM in 0.14s (system mean cpu 55%, single max cpu 100%), peaked 3.6 MiB above final usage, current RAM usage now 232.1 MiB


In [52]:
df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Sport'].map(dictio_sport_poids))


df_complete.info()

<class 'pandas.core.frame.DataFrame'>
Index: 96574 entries, 1 to 282626
Data columns (total 18 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Name            96574 non-null  object 
 1   Sex             96574 non-null  object 
 2   Age             96574 non-null  float64
 3   Height          96415 non-null  float64
 4   Weight          94584 non-null  float64
 5   Team            96574 non-null  object 
 6   NOC             96574 non-null  object 
 7   Games           96574 non-null  object 
 8   Year            96574 non-null  int64  
 9   Season          96574 non-null  object 
 10  City            96574 non-null  object 
 11  Sport           96574 non-null  object 
 12  Event           87481 non-null  object 
 13  Medal           96574 non-null  object 
 14  Population      93728 non-null  float64
 15  PIB/habitant    90843 non-null  float64
 16  Score_medaille  96574 non-null  int64  
 17  Country         96574 non-null  obj

<ipython-input-52-f91662992dfc>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Sport'].map(dictio_sport_poids))


In [53]:
df_complete=df_complete[df_complete['Season']=='Summer']

df_complete

Name Sex   Age  Height  Weight  \
1                        A Lamusi   M  23.0  170.00    60.0   
31              Jyri Tapani Aalto   M  31.0  172.00    70.0   
33             Minna Maarit Aalto   F  34.0  159.00    55.5   
57           Timo Antero Aaltonen   M  31.0  189.00   130.0   
79      Ragnhild Margrethe Aamodt   F  27.0  163.00    82.0   
...                           ...  ..   ...     ...     ...   
282622         ZOLOTIC Anastasija   F  19.0    1.80    69.0   
282623               ZOU Jingyuan   M  23.0    1.58     NaN   
282624               ZOU Jingyuan   M  23.0    1.58     NaN   
282625           ZUBIMENDI Martin   M  22.0    1.80    70.0   
282626           ZVEREV Alexander   M  24.0    1.98    72.0   

                              Team  NOC        Games  Year  Season     City  \
1                            China  CHN  2012 Summer  2012  Summer   London   
31                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
33                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
57                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
79                          Norway  NOR  2008 Summer  2008  Summer  Beijing   
...                            ...  ...          ...   ...     ...      ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer    Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer    Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer    Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer    Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer    Tokyo   

                      Sport                         Event         Medal  \
1                      Judo  Judo Men's Extra-Lightweight  Pas_medaille   
31                Badminton       Badminton Men's Singles  Pas_medaille   
33                  Sailing    Sailing Women's Windsurfer  Pas_medaille   
57                Athletics      Athletics Men's Shot Put  Pas_medaille   
79                 Handball     Handball Women's Handball          Gold   
...                     ...                           ...           ...   
282622            Taekwondo                   Women -57kg          Gold   
282623  Artistic Gymnastics                    Men's Team        Bronze   
282624  Artistic Gymnastics           Men's Parallel Bars          Gold   
282625             Football                           Men        Silver   
282626               Tennis                 Men's Singles          Gold   

          Population  PIB/habitant  Score_medaille        Country  
1       1.371220e+09   8027.683810               0          China  
31      5.482013e+06  42311.036231               0        Finland  
33      5.482013e+06  42311.036231               0        Finland  
57      5.482013e+06  42311.036231               0        Finland  
79      5.195921e+06  74400.369777               3         Norway  
...              ...           ...             ...            ...  
282622  3.214188e+08  56115.718426               3  United States  
282623  1.371220e+09   8027.683810               1          China  
282624  1.371220e+09   8027.683810               3          China  
282625  4.641827e+07  25831.582305               2          Spain  
282626  8.141314e+07  41313.313995               3        Germany  

[78790 rows x 18 columns]

In [53] used 0.8 MiB RAM in 0.15s (system mean cpu 22%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 229.3 MiB


## Moyenne par pays pour les sports qu'on ne connait pas 


In [54]:
table_pays_1=df_complete[df_complete['Weight'].isna()==False].groupby(['Country'])['Weight'].mean().apply(m.ceil).reset_index()

liste_pays_1=table_pays_1['Country'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids_1=table_pays_1['Weight'].tolist()



dictio_pays_poids_1=dict(zip(liste_pays_1,liste_poids_1))
df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Country'].map(dictio_pays_poids_1))

In [54] used 4.3 MiB RAM in 0.13s (system mean cpu 22%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 233.6 MiB


In [55]:
table_pays_2=df_complete[df_complete['Height'].isna()==False].groupby(['Country'])['Height'].mean().apply(m.ceil).reset_index()

liste_pays_2=table_pays_2['Country'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids_2=table_pays_2['Height'].tolist()



dictio_pays_poids_2=dict(zip(liste_pays_2,liste_poids_2))
df_complete['Height'] = df_complete['Height'].fillna(df_complete['Country'].map(dictio_pays_poids_2))

#df_complete.isna().sum()

In [55] used 0.6 MiB RAM in 0.14s (system mean cpu 27%, single max cpu 100%), peaked 8.3 MiB above final usage, current RAM usage now 234.2 MiB


In [56]:
#df_complete.to_csv('data/df_merge_FLO.csv', index=False)

In [56] used 0.0 MiB RAM in 0.11s (system mean cpu 41%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.2 MiB


## Web Scrapping FLo 

In [57]:
from bs4 import BeautifulSoup
import requests

In [57] used -48.3 MiB RAM in 1.04s (system mean cpu 11%, single max cpu 80%), peaked 44.1 MiB above final usage, current RAM usage now 185.9 MiB


In [58]:
url='https://www.topendsports.com/events/summer/hosts/list.htm'
response=requests.get(url)

In [58] used -4.1 MiB RAM in 1.19s (system mean cpu 36%, single max cpu 91%), peaked 8.8 MiB above final usage, current RAM usage now 181.8 MiB


In [59]:
url_2='https://en.wikipedia.org/wiki/List_of_Olympic_Games_host_cities'

In [59] used 0.0 MiB RAM in 0.10s (system mean cpu 46%, single max cpu 82%), peaked 0.0 MiB above final usage, current RAM usage now 181.8 MiB


In [60]:
response_2=requests.get(url_2)


In [60] used 1.1 MiB RAM in 0.24s (system mean cpu 40%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 183.0 MiB


In [61]:
soup_2=BeautifulSoup(response_2.text,'html.parser')


In [61] used 0.9 MiB RAM in 0.25s (system mean cpu 42%, single max cpu 86%), peaked 0.0 MiB above final usage, current RAM usage now 183.9 MiB


## On scrapp les pays organisateurs 

In [62]:
L_pays=[]
L_annees=[]
elements=soup_2.findAll('span',class_='datasortkey')

for element in elements:
    pays=element.find('a').text
    L_pays.append(pays)

L_pays=L_pays[42:58]




In [62] used 0.0 MiB RAM in 0.12s (system mean cpu 36%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 183.9 MiB


In [63]:
L_annees_2=[2000 + i  for i in range(0,34,2)]
L_annees_2.pop(-2)

L_annees_2

[2000,
 2002,
 2004,
 2006,
 2008,
 2010,
 2012,
 2014,
 2016,
 2018,
 2020,
 2022,
 2024,
 2026,
 2028,
 2032]

In [63] used -11.7 MiB RAM in 0.11s (system mean cpu 30%, single max cpu 100%), peaked 11.7 MiB above final usage, current RAM usage now 172.2 MiB


In [64]:
L_pays_annee=[]
for i in range(len(L_pays)):
    L_pays_annee.append((L_pays[i],L_annees_2[i]))

df_hote=pd.DataFrame(L_pays_annee,columns=['Host_country','Year'])

df_hote




Host_country  Year
0        Australia  2000
1    United States  2002
2           Greece  2004
3            Italy  2006
4            China  2008
5           Canada  2010
6   United Kingdom  2012
7           Russia  2014
8           Brazil  2016
9      South Korea  2018
10           Japan  2020
11           China  2022
12          France  2024
13           Italy  2026
14   United States  2028
15       Australia  2032

In [64] used 0.3 MiB RAM in 0.12s (system mean cpu 46%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 172.5 MiB


In [65]:
df_complete

Name Sex   Age  Height  Weight  \
1                        A Lamusi   M  23.0  170.00    60.0   
31              Jyri Tapani Aalto   M  31.0  172.00    70.0   
33             Minna Maarit Aalto   F  34.0  159.00    55.5   
57           Timo Antero Aaltonen   M  31.0  189.00   130.0   
79      Ragnhild Margrethe Aamodt   F  27.0  163.00    82.0   
...                           ...  ..   ...     ...     ...   
282622         ZOLOTIC Anastasija   F  19.0    1.80    69.0   
282623               ZOU Jingyuan   M  23.0    1.58    69.0   
282624               ZOU Jingyuan   M  23.0    1.58    69.0   
282625           ZUBIMENDI Martin   M  22.0    1.80    70.0   
282626           ZVEREV Alexander   M  24.0    1.98    72.0   

                              Team  NOC        Games  Year  Season     City  \
1                            China  CHN  2012 Summer  2012  Summer   London   
31                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
33                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
57                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
79                          Norway  NOR  2008 Summer  2008  Summer  Beijing   
...                            ...  ...          ...   ...     ...      ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer    Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer    Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer    Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer    Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer    Tokyo   

                      Sport                         Event         Medal  \
1                      Judo  Judo Men's Extra-Lightweight  Pas_medaille   
31                Badminton       Badminton Men's Singles  Pas_medaille   
33                  Sailing    Sailing Women's Windsurfer  Pas_medaille   
57                Athletics      Athletics Men's Shot Put  Pas_medaille   
79                 Handball     Handball Women's Handball          Gold   
...                     ...                           ...           ...   
282622            Taekwondo                   Women -57kg          Gold   
282623  Artistic Gymnastics                    Men's Team        Bronze   
282624  Artistic Gymnastics           Men's Parallel Bars          Gold   
282625             Football                           Men        Silver   
282626               Tennis                 Men's Singles          Gold   

          Population  PIB/habitant  Score_medaille        Country  
1       1.371220e+09   8027.683810               0          China  
31      5.482013e+06  42311.036231               0        Finland  
33      5.482013e+06  42311.036231               0        Finland  
57      5.482013e+06  42311.036231               0        Finland  
79      5.195921e+06  74400.369777               3         Norway  
...              ...           ...             ...            ...  
282622  3.214188e+08  56115.718426               3  United States  
282623  1.371220e+09   8027.683810               1          China  
282624  1.371220e+09   8027.683810               3          China  
282625  4.641827e+07  25831.582305               2          Spain  
282626  8.141314e+07  41313.313995               3        Germany  

[78790 rows x 18 columns]

In [65] used 0.1 MiB RAM in 0.13s (system mean cpu 31%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 172.5 MiB


In [66]:
df_complete=pd.merge(df_complete,df_hote,on='Year',how='left')

df_complete

Name Sex   Age  Height  Weight  \
0                       A Lamusi   M  23.0  170.00    60.0   
1              Jyri Tapani Aalto   M  31.0  172.00    70.0   
2             Minna Maarit Aalto   F  34.0  159.00    55.5   
3           Timo Antero Aaltonen   M  31.0  189.00   130.0   
4      Ragnhild Margrethe Aamodt   F  27.0  163.00    82.0   
...                          ...  ..   ...     ...     ...   
78785         ZOLOTIC Anastasija   F  19.0    1.80    69.0   
78786               ZOU Jingyuan   M  23.0    1.58    69.0   
78787               ZOU Jingyuan   M  23.0    1.58    69.0   
78788           ZUBIMENDI Martin   M  22.0    1.80    70.0   
78789           ZVEREV Alexander   M  24.0    1.98    72.0   

                             Team  NOC        Games  Year  Season     City  \
0                           China  CHN  2012 Summer  2012  Summer   London   
1                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
2                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
3                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
4                          Norway  NOR  2008 Summer  2008  Summer  Beijing   
...                           ...  ...          ...   ...     ...      ...   
78785    United States of America  USA   Tokyo 2020  2020  Summer    Tokyo   
78786  People's Republic of China  CHN   Tokyo 2020  2020  Summer    Tokyo   
78787  People's Republic of China  CHN   Tokyo 2020  2020  Summer    Tokyo   
78788                       Spain  ESP   Tokyo 2020  2020  Summer    Tokyo   
78789                     Germany  GER   Tokyo 2020  2020  Summer    Tokyo   

                     Sport                         Event         Medal  \
0                     Judo  Judo Men's Extra-Lightweight  Pas_medaille   
1                Badminton       Badminton Men's Singles  Pas_medaille   
2                  Sailing    Sailing Women's Windsurfer  Pas_medaille   
3                Athletics      Athletics Men's Shot Put  Pas_medaille   
4                 Handball     Handball Women's Handball          Gold   
...                    ...                           ...           ...   
78785            Taekwondo                   Women -57kg          Gold   
78786  Artistic Gymnastics                    Men's Team        Bronze   
78787  Artistic Gymnastics           Men's Parallel Bars          Gold   
78788             Football                           Men        Silver   
78789               Tennis                 Men's Singles          Gold   

         Population  PIB/habitant  Score_medaille        Country  \
0      1.371220e+09   8027.683810               0          China   
1      5.482013e+06  42311.036231               0        Finland   
2      5.482013e+06  42311.036231               0        Finland   
3      5.482013e+06  42311.036231               0        Finland   
4      5.195921e+06  74400.369777               3         Norway   
...             ...           ...             ...            ...   
78785  3.214188e+08  56115.718426               3  United States   
78786  1.371220e+09   8027.683810               1          China   
78787  1.371220e+09   8027.683810               3          China   
78788  4.641827e+07  25831.582305               2          Spain   
78789  8.141314e+07  41313.313995               3        Germany   

         Host_country  
0      United Kingdom  
1           Australia  
2           Australia  
3           Australia  
4               China  
...               ...  
78785           Japan  
78786           Japan  
78787           Japan  
78788           Japan  
78789           Japan  

[78790 rows x 19 columns]

In [66] used 3.4 MiB RAM in 0.14s (system mean cpu 35%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 176.0 MiB


In [67]:
compare_col=lambda x : 1 if x['Country']==x['Host_country'] else 0 




In [67] used -0.1 MiB RAM in 0.11s (system mean cpu 30%, single max cpu 100%), peaked 0.1 MiB above final usage, current RAM usage now 175.9 MiB


In [68]:
df_complete['Participation_own_country']=df_complete.apply(compare_col,axis=1)

df_complete[df_complete['Participation_own_country']==1]

Name Sex   Age  Height  Weight       Team  NOC  \
38          Giorgos Abalof   M  22.0  178.00    70.0     Greece  GRE   
40         Georgios Abaris   M  22.0  190.00    77.0     Greece  GRE   
49       Georgia Abatzidou   F  35.0  155.00    43.0     Greece  GRE   
156               Ali Abdo   M  19.0  169.00    74.0  Australia  AUS   
268    Diana Monteiro Abla   F  21.0  175.00    75.0     Brazil  BRA   
...                    ...  ..   ...     ...     ...        ...  ...   
78724        YANAGITA Yuki   M  33.0    1.88    66.0      Japan  JPN   
78741     YOSHIDA Masataka   M  28.0    1.73    66.0      Japan  JPN   
78742      YOSHIDA Tsukasa   F  26.0  172.00    79.0      Japan  JPN   
78743      YOSHIDA Tsukasa   F  26.0  172.00    79.0      Japan  JPN   
78744      YOSOZUMI Sakura   F  19.0    2.00    66.0      Japan  JPN   

             Games  Year  Season            City              Sport  \
38     2004 Summer  2004  Summer          Athina            Fencing   
40     2004 Summer  2004  Summer          Athina           Football   
49     2004 Summer  2004  Summer          Athina          Athletics   
156    2000 Summer  2000  Summer          Sydney          Wrestling   
268    2016 Summer  2016  Summer  Rio de Janeiro         Water Polo   
...            ...   ...     ...             ...                ...   
78724   Tokyo 2020  2020  Summer           Tokyo  Baseball/Softball   
78741   Tokyo 2020  2020  Summer           Tokyo  Baseball/Softball   
78742   Tokyo 2020  2020  Summer           Tokyo               Judo   
78743   Tokyo 2020  2020  Summer           Tokyo               Judo   
78744   Tokyo 2020  2020  Summer           Tokyo      Skateboarding   

                                           Event         Medal   Population  \
38                Fencing Men's epee, Individual  Pas_medaille   10823732.0   
40                       Football Men's Football  Pas_medaille   10823732.0   
49                    Athletics Women's Marathon  Pas_medaille   10823732.0   
156    Wrestling Men's Welterweight, Greco-Roman  Pas_medaille   23781169.0   
268                Water Polo Women's Water Polo  Pas_medaille  207847528.0   
...                                          ...           ...          ...   
78724                                   Baseball          Gold  126958472.0   
78741                                   Baseball          Gold  126958472.0   
78742                               Women -57 kg        Bronze  126958472.0   
78743                                 Mixed Team        Silver  126958472.0   
78744                               Women's Park          Gold  126958472.0   

       PIB/habitant  Score_medaille    Country Host_country  \
38     18002.230578               0     Greece       Greece   
40     18002.230578               0     Greece       Greece   
49     18002.230578               0     Greece       Greece   
156    56310.962993               0  Australia    Australia   
268     8538.589975               0     Brazil       Brazil   
...             ...             ...        ...          ...   
78724  32477.215145               3      Japan        Japan   
78741  32477.215145               3      Japan        Japan   
78742  32477.215145               1      Japan        Japan   
78743  32477.215145               2      Japan        Japan   
78744  32477.215145               3      Japan        Japan   

       Participation_own_country  
38                             1  
40                             1  
49                             1  
156                            1  
268                            1  
...                          ...  
78724                          1  
78741                          1  
78742                          1  
78743                          1  
78744                          1  

[3899 rows x 20 columns]

In [68] used 25.0 MiB RAM in 0.56s (system mean cpu 34%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 200.9 MiB


In [69]:
table_sports=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sports

Sport   Name
0          3x3 Basketball     56
1                 Archery   1139
2     Artistic Gymnastics    191
3       Artistic Swimming    100
4               Athletics  13711
5               Badminton   1135
6                Baseball    574
7       Baseball/Softball    234
8              Basketball   1715
9        Beach Volleyball    570
10                 Boxing   1711
11           Canoe Slalom     80
12           Canoe Sprint    243
13               Canoeing   2177
14                Cycling   3192
15  Cycling BMX Freestyle     16
16     Cycling BMX Racing     42
17  Cycling Mountain Bike     74
18           Cycling Road    189
19          Cycling Track    224
20                 Diving   1114
21             Equestrian    232
22          Equestrianism   1711
23                Fencing   1937
24               Football   2807
25                   Golf    240
26             Gymnastics   5000
27               Handball   2034
28                 Hockey   2272
29                   Judo   2330
30                 Karate     79
31      Marathon Swimming     47
32      Modern Pentathlon    391
33    Rhythmic Gymnastics    543
34                 Rowing   3282
35           Rugby Sevens    598
36                Sailing   2301
37               Shooting   3190
38          Skateboarding     79
39               Softball    358
40         Sport Climbing     36
41                Surfing     40
42               Swimming   9124
43  Synchronized Swimming    586
44           Table Tennis   1425
45              Taekwondo    728
46                 Tennis   1508
47  Trampoline Gymnastics     28
48           Trampolining    152
49              Triathlon    637
50             Volleyball   1677
51             Water Polo   1530
52          Weightlifting   1446
53              Wrestling   1955

In [69] used -2.1 MiB RAM in 0.12s (system mean cpu 47%, single max cpu 100%), peaked 2.1 MiB above final usage, current RAM usage now 198.8 MiB


## Dictionnaire group sport

On va essayer de scrapper les sports individuels et les collectifs 


In [70]:
url_sports='https://www.britannica.com/topic/list-of-sports-2038581#ref330452'

In [70] used -4.9 MiB RAM in 0.10s (system mean cpu 32%, single max cpu 100%), peaked 4.9 MiB above final usage, current RAM usage now 193.9 MiB


In [71]:
response_sp=requests.get(url_sports)

In [71] used -18.7 MiB RAM in 0.99s (system mean cpu 24%, single max cpu 100%), peaked 9.5 MiB above final usage, current RAM usage now 175.2 MiB


In [72]:
soup_sports=BeautifulSoup(response_sp.text,'html.parser')


In [72] used 21.9 MiB RAM in 0.24s (system mean cpu 24%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 197.1 MiB


In [73]:
sports=soup_sports.findAll('div')
L_sports_indiv=[]
for sport in sports:
    a=sport.find('a',class_='md-crosslink')
    if a is not None:
        L_sports_indiv.append(a.text)
   
    
L_sports_collec=[]

sports_co=soup_sports.findAll('section',id='ref330452')

for sport in sports_co:
    b=sport.find('a',class_='md-crosslink')
    print(b)
   
        
#sports

NameError: name 'L' is not defined

In [73] used -7.3 MiB RAM in 0.20s (system mean cpu 24%, single max cpu 100%), peaked 7.3 MiB above final usage, current RAM usage now 189.7 MiB


## Pays passé communiste 

In [74]:
url_soviet='https://en.wikipedia.org/wiki/Post-Soviet_states'

In [74] used -174.4 MiB RAM in 0.11s (system mean cpu 23%, single max cpu 38%), peaked 0.0 MiB above final usage, current RAM usage now 15.4 MiB


In [77]:
response_soviet=requests.get(url_soviet)

In [77] used 6.7 MiB RAM in 0.19s (system mean cpu 34%, single max cpu 56%), peaked 0.0 MiB above final usage, current RAM usage now 45.9 MiB


In [79]:
soup_soviet=BeautifulSoup(response_soviet.text,'html.parser')

In [79] used 14.3 MiB RAM in 0.35s (system mean cpu 44%, single max cpu 78%), peaked 0.0 MiB above final usage, current RAM usage now 60.5 MiB


In [104]:
L_pays_soviet=[]

pays_soviet=soup_soviet.findAll('b')
#print(pays_soviet)
for pays in pays_soviet:
    b=pays.find('a')
    if b is not None:
        L_pays_soviet.append(b.text)
L_pays_soviet=L_pays_soviet[:15]

L_pays_soviet

['Kazakhstan',
 'Kyrgyzstan',
 'Tajikistan',
 'Turkmenistan',
 'Uzbekistan',
 'Belarus',
 'Moldova',
 'Russia',
 'Ukraine',
 'Estonia',
 'Latvia',
 'Lithuania',
 'Armenia',
 'Azerbaijan',
 'Georgia']

In [104] used -0.0 MiB RAM in 0.13s (system mean cpu 32%, single max cpu 55%), peaked 0.0 MiB above final usage, current RAM usage now 27.4 MiB


In [108]:
complete_urss_regime=lambda x : 1 if x in L_pays_soviet else 0

In [108] used 0.9 MiB RAM in 0.10s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 43.0 MiB


In [110]:
df_complete['Soviet_past']=df_complete['Country'].apply(complete_urss_regime)



Name Sex   Age  Height  Weight  \
13                                Moonika Aava   F  24.0  168.00    65.0   
14                                Moonika Aava   F  28.0  168.00    65.0   
25                           Patimat Abakarova   F  21.0  165.00    49.0   
26     Mariya Vasilyevna Abakumova (-Tarabina)   F  22.0  179.00    80.0   
27     Mariya Vasilyevna Abakumova (-Tarabina)   F  26.0  179.00    80.0   
...                                        ...  ..   ...     ...     ...   
78528                        TUSHISHVILI Guram   M  26.0  172.00    79.0   
78532                       TYNYBEKOVA Aisuluu   F  28.0  171.00    77.0   
78714                        YAKHNO Yelyzaveta   F  23.0    1.75    70.0   
78761                           ZARETSKA Iryna   F  25.0  168.00    71.0   
78783                     ZHUMANAZAROVA Meerim   F  22.0  171.00    77.0   

             Team  NOC        Games  Year  Season  ...              Sport  \
13        Estonia  EST  2004 Summer  2004  Summer  ...          Athletics   
14        Estonia  EST  2008 Summer  2008  Summer  ...          Athletics   
25     Azerbaijan  AZE  2016 Summer  2016  Summer  ...          Taekwondo   
26         Russia  RUS  2008 Summer  2008  Summer  ...          Athletics   
27         Russia  RUS  2012 Summer  2012  Summer  ...          Athletics   
...           ...  ...          ...   ...     ...  ...                ...   
78528     Georgia  GEO   Tokyo 2020  2020  Summer  ...               Judo   
78532  Kyrgyzstan  KGZ   Tokyo 2020  2020  Summer  ...          Wrestling   
78714     Ukraine  UKR   Tokyo 2020  2020  Summer  ...  Artistic Swimming   
78761  Azerbaijan  AZE   Tokyo 2020  2020  Summer  ...             Karate   
78783  Kyrgyzstan  KGZ   Tokyo 2020  2020  Summer  ...          Wrestling   

                                 Event         Medal   Population  \
13     Athletics Women's Javelin Throw  Pas_medaille    1311998.0   
14     Athletics Women's Javelin Throw  Pas_medaille    1311998.0   
25         Taekwondo Women's Flyweight        Bronze    9651349.0   
26     Athletics Women's Javelin Throw        Silver  144096812.0   
27     Athletics Women's Javelin Throw  Pas_medaille  144096812.0   
...                                ...           ...          ...   
78528                      Men +100 kg        Silver    3679000.0   
78532           Women's Freestyle 62kg        Silver    5957000.0   
78714                             Team        Bronze   45198200.0   
78761             Women's Kumite +61kg        Silver    9651349.0   
78783           Women's Freestyle 68kg        Bronze    5957000.0   

       PIB/habitant  Score_medaille     Country    Host_country  \
13     17118.504200               0     Estonia          Greece   
14     17118.504200               0     Estonia           China   
25      5496.344640               1  Azerbaijan          Brazil   
26      9092.580536               2      Russia           China   
27      9092.580536               0      Russia  United Kingdom   
...             ...             ...         ...             ...   
78528   3795.973308               2     Georgia           Japan   
78532   1103.215352               2  Kyrgyzstan           Japan   
78714   2114.954716               1     Ukraine           Japan   
78761   5496.344640               2  Azerbaijan           Japan   
78783   1103.215352               1  Kyrgyzstan           Japan   

      Participation_own_country  Soviet_past  
13                            0            1  
14                            0            1  
25                            0            1  
26                            0            1  
27                            0            1  
...                         ...          ...  
78528                         0            1  
78532                         0            1  
78714                         0            1  
78761                         0            1  
78783                         0            1  

[8925

In [110] used -19.5 MiB RAM in 0.17s (system mean cpu 32%, single max cpu 57%), peaked 0.0 MiB above final usage, current RAM usage now 93.9 MiB


In [111]:
df_complete.isna().sum()

Name                            0
Sex                             0
Age                             0
Height                          0
Weight                          0
Team                            0
NOC                             0
Games                           0
Year                            0
Season                          0
City                            0
Sport                           0
Event                        9093
Medal                           0
Population                   2609
PIB/habitant                 5315
Score_medaille                  0
Country                         0
Host_country                    0
Participation_own_country       0
Soviet_past                     0
dtype: int64

In [111] used -47.0 MiB RAM in 0.17s (system mean cpu 34%, single max cpu 61%), peaked 0.0 MiB above final usage, current RAM usage now 46.9 MiB


In [113]:
df_pop_na=df_complete[df_complete['Population'].isna()]

#df_pop_na.groupby('Country')['Name'].count().reset_index()

Country  Name
0                      Cook Islands    33
1                           Eritrea    56
2       Individual Olympic Athletes    17
3                          Kiribati    14
4                            Kosovo    19
5                             Liban     6
6                  Marshall Islands    16
7                        Montenegro   124
8             Netherlands Antilles*    15
9   Palestine, Occupied Territories    25
10                          Romania   988
11                           Serbia   445
12                        Singapour   172
13                      South Sudan     5
14                           Taiwan   461
15                         Trinidad   204
16                           Tuvalu     9

In [113] used -1.3 MiB RAM in 0.13s (system mean cpu 30%, single max cpu 54%), peaked 0.0 MiB above final usage, current RAM usage now 44.3 MiB


## Scrap Population et GPA 

In [132]:
import re

In [132] used -3.8 MiB RAM in 0.11s (system mean cpu 32%, single max cpu 56%), peaked 0.0 MiB above final usage, current RAM usage now 13.5 MiB


In [114]:
url_pop_gpa='https://www.worldometers.info/gdp/gdp-by-country/'

In [114] used -29.2 MiB RAM in 0.12s (system mean cpu 34%, single max cpu 62%), peaked 0.0 MiB above final usage, current RAM usage now 15.1 MiB


In [115]:
response_pop_gpa=requests.get(url_pop_gpa)


In [115] used 12.1 MiB RAM in 0.64s (system mean cpu 39%, single max cpu 89%), peaked 0.0 MiB above final usage, current RAM usage now 27.2 MiB


In [117]:
soup_pop_gpa=BeautifulSoup(response_pop_gpa.text,'html.parser')


In [117] used 6.3 MiB RAM in 0.18s (system mean cpu 40%, single max cpu 62%), peaked 0.0 MiB above final usage, current RAM usage now 45.0 MiB


In [125]:
L_pays=[]

all_countries=soup_pop_gpa.findAll('td')

for country in all_countries:
    a=country.find('a')
    if a is not None:
        L_pays.append(a.text)
len(L_pays)

177

In [125] used 1.0 MiB RAM in 0.13s (system mean cpu 42%, single max cpu 73%), peaked 0.0 MiB above final usage, current RAM usage now 21.8 MiB


In [ ]:
b=re.findall(r'\d+\,?\d*|\d+\.\d+', info.text)

In [158]:
all_info=soup_pop_gpa.findAll('td',attrs={'style':'font-weight: bold; text-align:right;'})
#print(all_info)
numbers=[]
for i,info in enumerate(all_info):
    b=re.findall(r'\d+\b', info.text)
    b=''.join(b)
    numbers.append(b)
   
 
    #b=[number.replace(',','.') for number in b]
print(len(numbers))
population=[]
for i in range(4,len(numbers),4):
    population.append(numbers[i])
population

['25', '462', '700', '000', '000']
['25', '463']
['2', '06']
['338', '289', '857']
['75', '269']
['25', '32']
['17', '963', '200', '000', '000']
['17', '963']
['2', '99']
['1', '425', '887', '337']
['12', '598']
['17', '86']
['4', '231', '140', '000', '000']
['4', '231']
['1', '03']
['123', '951', '692']
['34', '135']
['4', '21']
['4', '072', '190', '000', '000']
['4', '072']
['1', '79']
['83', '369', '843']
['48', '845']
['4', '05']
['3', '385', '090', '000', '000']
['3', '385']
['7', '00']
['1', '417', '173', '173']
['2', '389']
['3', '37']
['3', '070', '670', '000', '000']
['3', '071']
['4', '10']
['67', '508', '936']
['45', '485']
['3', '05']
['2', '782', '910', '000', '000']
['2', '783']
['2', '56']
['64', '626', '628']
['43', '061']
['2', '77']
['2', '240', '420', '000', '000']
['2', '240']
['2', '07']
['144', '713', '314']
['15', '482']
['2', '23']
['2', '139', '840', '000', '000']
['2', '140']
['3', '40']
['38', '454', '327']
['55', '646']
['2', '13']
['2', '010', '430', '000',